components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -224,6 +252,78 @@ public Action getActionByActionId(String actionId, Integer tenantId) throws Acti
         }
     }
 
+    @Override
+    public Action updateActionEndpointAuthProperties(String actionId, AuthType authentication, int tenantId)
+            throws ActionMgtException {
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true);
+        try {
+            Map<String, String> nonSecretEndpointProperties = authentication.getProperties().stream()
+                    .filter(property -> !property.getIsConfidential())
+                    .collect(Collectors.toMap(AuthProperty::getName, AuthProperty::getValue));
+            // Update non-secret endpoint properties.
+            updateActionEndpointProperties(dbConnection, actionId, nonSecretEndpointProperties, tenantId);
+            // Encrypt and update non-secret endpoint properties.
```

Comment:
Shouldn't this comment be "Encrypt and update secret endpoint properties."

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5795/commits/d2420ad5643697c5521872bf43d17fe17dd12580